pkg/chartutil/chartfile.go

```diff
@@ -91,3 +91,13 @@ func IsChartDir(dirName string) (bool, error) {
 
 	return true, nil
 }
+
+func LoadChartFileForTypeCheck(filename string) (map[string]interface{}, error) {
```

Comment:
Why is this a public function? That is, a new function to be part of the public API under the semver guarantee? Would this be better as a private function that is part of the linting of the chartfile?

If this should be a public function and part of the API it needs a comment describing when you would use it and why it would be used instead of LoadChartfile. Since the name refers to type checking someone might wonder how to do that.

On my first pass, I would do this as a private function in pkg/link/rules/chartfile.go.

Reply:
Makes sense. I'll change it into a private method and also put a comment regarding what it does